README.md

```diff
@@ -26,38 +26,144 @@ guarantee these changes will be backwards-compatible.
   - [Design Space](https://docs.google.com/document/d/1TpQlQYovCoX9FkpgsoxzdvZplghudHAiQOame30A-v8/edit)
-- [Frequently Asked Questions](FAQ.md)
+- [Frequently Asked Questions](docs/FAQ.md)
 
-## Usage
+## Setup
 
 Get the tool via
 
 ```sh
 $ go get -u github.com/golang/dep/cmd/dep
 ```
 
-Typical usage on a new repo might be
+To start managing dependencies using dep, run the following from your project root directory:
 
 ```sh
 $ dep init
 $ dep ensure -update
 ```
 
-To update a dependency to a new version, you might run
+`dep init` will do the following:
+
+1. Look for [existing dependency management files](docs/FAQ.md#what-external-tools-are-supported) to convert
+1. Check if your dependencies use dep
+1. Identify your dependencies
+1. Back up your existing `vendor/` directory (if you have one) to
+`_vendor-TIMESTAMP/`
+1. Pick the highest compatible version for each dependency
+1. Generate [`Gopkg.toml`](Gopkg.toml.md) and `Gopkg.lock` files
+1. Install the dependencies in `vendor/`
+
+## Usage
+
+There is one main subcommand you will use: `dep ensure`. `ensure` synchronizes
+your dependencies in `vendor/` to make sure they match what's in your `import`s
+and `Gopkg.toml`. `dep ensure` is safe to run early and often. See the help text
+for more detailed usage instructions.
 
 ```sh
-$ dep ensure github.com/pkg/errors@^0.8.0
+$ dep help ensure
 ```
 
-See the help text for more detailed usage instructions.
+### Installing dependencies
+
+(if your `vendor/` directory isn't [checked in with your code](](docs/FAQ.md#should-i-commit-my-vendor-directory)))
+
+```sh
+$ dep ensure
+```
+
+If a dependency already exists in your `vendor/` folder, dep will ensure it
+matches the constraints from the manifest. If the dependency is missing from
+`vendor/`, the latest version allowed by your manifest will be installed.
+
+### Adding a dependency
+
+1. Add the `import` in your source code file(s).
+1. Download via dep.
+
+    ```sh
+    $ dep ensure
+    ```
+
+### Changing dependencies
+
+If you want to:
+
+* Change the allowed `version`/`branch`/`revision`
+* Switch to using a fork
+
+for one or more dependencies, do the following:
+
+1. Modify your `Gopkg.toml`.
+1. Run
+
+    ```sh
+    $ dep ensure
+    ```
+
+### Checking the status of dependencies
+
+```sh
+$ dep status
+PROJECT                             CONSTRAINT     VERSION        REVISION  LATEST
+github.com/Masterminds/semver       branch 2.x     branch 2.x     139cc09   c2e7f6c
+github.com/Masterminds/vcs          ^1.11.0        v1.11.1        3084677   3084677
+github.com/armon/go-radix           *              branch master  4239b77   4239b77
+```
+
+### Updating dependencies
+
+(to the latest version allowed by the manifest)
+
+```sh
+$ dep ensure -update
+```
+
+### Removing dependencies
+
+1. Remove the `import`s and all usage from your code.
+1. Run
+
+    ```sh
+    $ dep ensure
+    ```
+
+1. Remove from `Gopkg.toml`, if it was in there.
+
+### Testing changes to a dependency
```

Comment:
This simple advice will work for quickly testing local changes, but not for actually keeping any of them in `Gopkg.lock` or `vendor` in any kind of portable way. It's probably worth noting that, as well as that we may not have a really good solution for this until after toolchain integration  (see [gist](https://gist.github.com/sdboyer/def9b138af448f39300cb078b0e94cc3))

It may be worth linking to [virtualgo](https://github.com/GetStream/vg), as well, as it offers one interim attempt at a solution.